src/core/worker.js

```diff
@@ -436,6 +437,44 @@ var WorkerMessageHandler = PDFJS.WorkerMessageHandler = {
       });
     });
 
+    handler.on('GetRawObject', function wphGetRawObject(data) {
+      function clone(obj) {
+        if (!obj || isBool(obj) || isNum(obj) || isString(obj)) {
+          return obj;
+        } else if (isName(obj)) {
+          return { name: obj.name };
+        } else if (isDict(obj) || isArray(obj)) {
+          return obj.map(clone);
```

Comment:
The `.map` property of a dict is the internal representation of a `Dict`. The original way of cloning the object should be restored.

The `.map` property of an array is a standard JavaScript method that iterates over the array and returns the result of calling the function (`clone` in your case) on the array element.

I suggested to use `.map` instead of your previous `for` loop, but you combined both cases (`isDict` and `isArray`), probably because of a misunderstanding of what `map` is.